packages/client/hmi-client/src/services/models/stratification-service.ts

```diff
@@ -39,28 +43,78 @@ export function generateAgeStrataModel(stateNames: string[]) {
 				output: [outerState.id, innerState.id],
 				properties: {
 					name: `c&#832${i + 1}&#832${j + 1}`,
-					description: 'Infective interaction between individuals'
+					description: 'Infective interaction between individuals.'
 				}
 			});
 		})
 	);
+	const typeMap: string[][] = states
+		.map((state) => [state.id, 'Pop'])
+		.concat(transitions.map((transition) => [transition.id, 'Strata']));
+	const semantics: ModelSemantics = {
+		ode: {
+			rates: []
+		},
+		typing: {
+			type_system: {
+				states: [
+					{
+						id: 'Pop',
+						name: 'Pop',
+						description: 'Compartment of individuals in a human population.'
+					}
+				],
+				transitions: [
+					{
+						id: 'Strata',
+						input: ['Pop'],
+						output: ['Pop'],
+						properties: {
+							name: 'Strata',
+							description:
+								'1-to-1 process that represents a change in the demographic division of a human individual.'
+						}
+					}
+				]
+			},
+			type_map: typeMap
+		}
+	};
 	return {
+		id: '',
+		framework: '',
 		name: 'Age-contact strata model',
 		description: 'Age-contact strata model',
+		schema:
+			'https://raw.githubusercontent.com/DARPA-ASKEM/Model-Representations/petrinet_v0.1/petrinet/petrinet_schema.json',
+		schema_name: 'petrinet',
+		model_version: '0.1',
```

Comment:
This amr creation is getting repeated over and over again, I would maybe suggest that we have a function in petrinet-service that builds a basic model skeleton, something like 

```
const amr = petrinetService.newModel(id, name, description ....)
```

Reply:
I agree. If you don't mind, I can address that in the next pr.